hoomd/md/methods/methods.py

```diff
@@ -651,62 +517,7 @@ def barostat_energy(self):
         return self._cpp_obj.getBarostatEnergy(self._simulation.timestep)
 
 
-class NVE(Method):
-    r"""Constant volume, constant energy dynamics.
-
-    Args:
-        filter (hoomd.filter.filter_like): Subset of particles on which to
-            apply this method.
-
-    `NVE` integrates integrates translational and rotational degrees of freedom
-    in the microcanonical ensemble. The equations of motion are derived from the
-    hamiltonian:
-
-    .. math::
-
-        H = U + K_\mathrm{translational} + K_\mathrm{rotational}
-
-    `NVE` numerically integrates the translational degrees of freedom
-    using Velocity-Verlet and the rotational degrees of freedom with a scheme
-    based on `Kamberaj 2005`_.
-
-    Examples::
-
-        nve = hoomd.md.methods.NVE(filter=hoomd.filter.All())
-        integrator = hoomd.md.Integrator(dt=0.005, methods=[nve], forces=[lj])
-
-    .. _Kamberaj 2005: http://dx.doi.org/10.1063/1.1906216
-
-    Attributes:
-        filter (hoomd.filter.filter_like): Subset of particles on which to
-            apply this method.
-    """
-
-    def __init__(self, filter):
-
-        # store metadata
-        param_dict = ParameterDict(filter=ParticleFilter,)
-        param_dict["filter"] = filter
-
-        # set defaults
-        self._param_dict.update(param_dict)
-
-    def _attach_hook(self):
-
-        sim = self._simulation
-        # initialize the reflected c++ class
-        if isinstance(sim.device, hoomd.device.CPU):
-            self._cpp_obj = _md.TwoStepNVE(sim.state._cpp_sys_def,
-                                           sim.state._get_group(self.filter))
-        else:
-            self._cpp_obj = _md.TwoStepNVEGPU(sim.state._cpp_sys_def,
-                                              sim.state._get_group(self.filter))
-
-        # Attach param_dict and typeparam_dict
-        super()._attach_hook()
-
-
-class DisplacementCapped(NVE):
+class DisplacementCapped(ConstantVolume):
```

Comment:
`ConstantVolume.[cc, cu]` are based on the original NVT code which lacked the `movement_limit` implementation. We either need to deprecate and remove this, or add support in to `ConstantVolume`. Some users like to use this option when relaxing initial configurations.

See 
* https://github.com/glotzerlab/hoomd-blue/blob/3539c5ea43502b184a7a02dbdc42524ab1fcf08b/hoomd/md/TwoStepNVE.cc#L92-L103 
* https://github.com/glotzerlab/hoomd-blue/blob/3539c5ea43502b184a7a02dbdc42524ab1fcf08b/hoomd/md/TwoStepNVEGPU.cu#L86-L92
* and #1363
for context.

Reply:
I'm adding support in constant volume. I have a question related to the current implementation. Is it necessary to enforce limits on velocity in both update steps, ie integrateStepOne and integrateStepTwo? In the first half-step, the velocity already gets clipped.

Reply:
Oh, does it? I missed that. The original implementation of the limit applied it only to the position displacement - so it was only applied in step one.

Reply:
I pushed a displacement capped version where capping is done on step 1 only @b-butler was there a particular reason why it should be in both steps?

Reply:
I don't believe so. If I did that it was probably just a mistake on my part.